feat: add team integration support in Agent UI

fioritonicolas opened the pull request:
**Summary**

  - Added support for team-based agent interactions in the Agent UI (Fixes #82 , #87 )
  - Implemented adaptive UI that automatically uses teams when available from the server and uses agents otherwise
      - So, when we run Playground like this: app = Playground(teams=[team_leader]).get_app(), it will load the TeamLeader in the Select.
  - Extended session management to work seamlessly with both agent and team sessions

  **Key Changes**

  - Added new TeamSelector component for selecting and displaying available teams
  - Implemented team API endpoints for fetching teams, sessions, and handling runs
  - Updated state management to track entity type (agent vs team) and handle context switching
  - Enhanced session management to load, display, and delete team-specific sessions
  - Improved error handling for team operations with better user feedback
  - Fixed session handling during streaming responses to properly maintain session state
  
  
**I really like what you’re doing with Agno, so I wanted to help out a bit here. I’m open to making any changes needed to adapt this feature to match your vision.**

This is how the UI looks:

<img width="290" alt="image" src="https://github.com/user-attachments/assets/82b3932d-54fd-4b92-8ad9-a7e5a5c1fdc3" />
  

  
  **Important**

  
  - To maintain session history when using a database, the team leader on the playground server must be started with a specific team_id. If it's not provided, a new team ID is generated each time the page is refreshed, causing messages to be saved under different team leader IDs. This makes it impossible to retrieve previous sessions, even with a database. I believe this behavior stems from how teams are initialized in playground.py.
  
 
```
FINANCE_TEAM_LEADER_ID = "80a9047b-9918-4db1-84df-1e6152f073ae"

team_leader = Team(
    name="Reasoning Finance Team Leader",
    mode="coordinate",
    team_id=FINANCE_TEAM_LEADER_ID,
    model=Claude(id="claude-3-7-sonnet-latest"),
    members=[
        web_agent,
        finance_agent,
    ],
    tools=[ReasoningTools(add_instructions=True)],
    instructions=[
        "Only output the final answer, no other text.",
        "Use tables to display data",
    ],
    markdown=True,
    show_members_responses=True,
    enable_agentic_context=True,
    add_datetime_to_instructions=True,

    storage=SqliteStorage(table_name="team_leader", db_file=agent_storage),
    success_criteria="The team has successfully completed the task."
)

app = Playground(teams=[team_leader]).get_app()
```

chenzikun commented:
@fioritonicolas  
I found there is bug where there has team and agent at same time.
I will publish my code base on your code, pls check!

here is the [pull request](https://github.com/fioritonicolas/agent-ui/pull/1)

Here is my snapshot!
1. where chose agent
![image](https://github.com/user-attachments/assets/44561aef-ac4b-480f-b02e-4998ddabc5df)
2. when chose team
![image](https://github.com/user-attachments/assets/c412854a-cbce-44c2-bbb2-294530ff6f3e)

m-breitbach commented:
I tested the branch in this PR and can confirm @chenzikun finding. It works on their fork, however, I find the UX not that good. How about putting all top-level teams and agents in one drop down list, and make a second list appear showing team members, whenever a team is selected. (This should theoretically be possible by `team_session_id`.

mhannani commented:
We need definitly that features. thank so your pull requests @fioritonicolas and @chenzikun !

sebastian-montero commented:
we need this!!!!

jimilibilibob commented:
We need this please 🙏

eaglespy21 commented:
Really need this!

fioritonicolas commented:
Hey @Ayush0054, of course, I’d love to get this finished and see the PR merged! 😄

The definitions you shared are very much in line with what I had in mind. I’ll sync with the latest main branch and start working on the updates you mentioned.

Thanks

fioritonicolas commented:
Hey @Ayush0054

I’ve implemented all the requested changes, give it a try! I’d love to hear your feedback. I’ve tested it with the finance team, and it’s working really well.

### **Mode Dropdown UI**
- Added a single **Mode** selector (`Agent` / `Team`) matching **app.agno.com**.
- **Agent mode** lists agents; **Team mode** lists teams.
- Dropdown is disabled unless both agents and teams exist, auto-defaults to the available mode.

### **Streaming Updates** (per team core events)
- Updated the stream handler to support team events:  
  `RunStarted` / `ResponseContent` / `Completed` / `Error` / `Cancelled`  
  `ToolCallStarted` / `Completed`  
  `ReasoningStarted` / `Step` / `Completed`
- Tool calls and reasoning steps now update incrementally during streaming.

### **Sessions**
- Session loader now uses `teamId` for team sessions.
- Sessions list loads, opens, and deletes for both agents and teams.
- New session is inserted on first response chunk when storage is enabled.

Some screenshots of the answer from the team:

<img width="1343" height="809" alt="image" src="https://github.com/user-attachments/assets/c82d052b-d65b-4159-85aa-21e5e8725ed8" />
<img width="1262" height="746" alt="image" src="https://github.com/user-attachments/assets/97cb2437-0621-49ed-b541-0043a9572017" />

Ayush0054 commented on src/store.ts:
```diff
@@ -49,8 +59,16 @@ interface PlaygroundStore {
   setSelectedModel: (model: string) => void
+  selectedTeamId: string | null
+  setSelectedTeamId: (teamId: string | null) => void
+  selectedEntityType: 'agent' | 'team' | null
```
really great work @fioritonicolas  ,
just few comments now
Why do we have both mode and entityType? I think just one would be enough — I’d prefer mode.

fioritonicolas replied:
I agree, and I don’t think we need to have both. I’ll adapt the change to use only mode. I’ll also initialize the state with mode: team, so if there’s a team in the playground, it will load it by default. If there’s no team, it will use the logic to show the mode as disabled and display the agents.

Ayush0054 commented on src/components/playground/Sidebar/EntitySelector.tsx:
```diff
@@ -0,0 +1,153 @@
+  const placeholder = mode === 'team' ? 'Select Team' : 'Select Agent'
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
```
why we are using setTimeout here 😅

fioritonicolas replied:
Ouch, sorry 😅  this timeout shouldn’t have made it into the PR, and the useEffect shouldn’t be here either. This was just some testing I was doing for the initialization logic we already have in initializePlayground and i forgot to delete it

Ayush0054 commented on src/components/playground/Sidebar/EntitySelector.tsx:
```diff
@@ -0,0 +1,153 @@
+  const placeholder = mode === 'team' ? 'Select Team' : 'Select Agent'
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
```
also whats the purpose of both useEffects here ?  please explain a bit. 😅

fioritonicolas replied:
Regarding the first one, it was just me testing and trying to understand how agent‑ui works.

For the second one, I thought of it as a safety net, but it feels a bit redundant because we handle and initialize in initializePlayground and in handleOnValueChange here in EntitySelector.

I still think it’s needed for edge cases mainly:

-  URL driven changes: when the user manually edits the URL or uses browser navigation
- Race conditions: when entities load after URL parameters are set

For example, if I refresh the URL with this team ID, without that useEffect I will lose the conversation history, and I’d have to refresh the base localhost:3000 for the UI to fetch everything again. With this useEffect i and same to perform that refresh

<img width="487" height="44" alt="image" src="https://github.com/user-attachments/assets/9613a128-88cf-4409-b3db-98d1158771ed" />

Let me know what you think

Ayush0054 replied:
Yes, agreed
 let’s keep it for now. Thanks for the detailed explanation.

fioritonicolas replied:
When merge? 😛 Let me know if you need anything extra

Ayush0054 replied:
hey @fioritonicolas, we’ll merge it soon — the team is currently testing it, and once that’s done, we’ll merge.

fioritonicolas commented:
@Ayush0054 Thanks very much, I really appreciate your kind words. I’ve replied to your comments and made the changes accordingly.

Ayush0054 commented:
> @Ayush0054 Thanks very much, I really appreciate your kind words. I’ve replied to your comments and made the changes accordingly.

thanks @fioritonicolas its an great contribution for agent-ui 🚀

pritipsingh commented:
Apologies for the delay and thanks for the contributions @fioritonicolas - we really appreciate it! 

We are able to stream well, but we do not yet persist the sessions created in this PR. We can use this API for this: `/v1/playground/teams/{team_id}/sessions` - take the `session_id` from that API and load the session. Although this is good at the moment, and I am sure the community will love this change :)

Thank You

fioritonicolas commented:
@pritipsingh Thanks. Okey i will investigate it and open another PR for that.
